wpilibj/src/main/java/edu/wpi/first/wpilibj/XboxController.java

```diff
@@ -93,170 +108,136 @@ public boolean getBumper(Hand hand) {
-  }
-
-  /**
    * Read the value of the back button on the controller.
    *
+   * @deprecated Use getButton() instead.
    * @return The state of the button.
    */
+  @Deprecated
   public boolean getBackButton() {
     return getRawButton(7);
   }
 
   /**
    * Read the value of the start button on the controller.
    *
+   * @deprecated Use getButton() instead.
    * @return The state of the button.
    */
+  @Deprecated
   public boolean getStartButton() {
     return getRawButton(8);
   }
 
-  @Override
-  public int getPOV(int pov) {
-    return m_ds.getStickPOV(getPort(), pov);
-  }
-
-  @Override
-  public int getPOVCount() {
-    return m_ds.getStickPOVCount(getPort());
-  }
-
-  @Override
-  public HIDType getType() {
-    return HIDType.values()[m_ds.getJoystickType(getPort())];
-  }
-
-  @Override
-  public String getName() {
-    return m_ds.getJoystickName(getPort());
-  }
-
-  @Override
-  public void setOutput(int outputNumber, boolean value) {
-    m_outputs = (m_outputs & ~(1 << (outputNumber - 1))) | ((value ? 1 : 0) << (outputNumber - 1));
-    HAL.setJoystickOutputs((byte) getPort(), m_outputs, m_leftRumble, m_rightRumble);
+  /**
+   * Whether the button was pressed since the last check. Button indexes begin at 1. The button
+   * type is mapped to a button index first.
+   *
+   * @param button The button index, beginning at 1.
+   * @return Whether the button was pressed since the last check.
+   */
+  boolean getButtonPressed(Button button) {
+    return getRawButtonPressed(getButtonIndex(button));
   }
 
-  @Override
-  public void setOutputs(int value) {
-    m_outputs = value;
-    HAL.setJoystickOutputs((byte) getPort(), m_outputs, m_leftRumble, m_rightRumble);
+  /**
+   * Whether the button was released since the last check. Button indexes begin at 1. The button
+   * type is mapped to a button index first.
+   *
+   * @param button The button index, beginning at 1.
+   * @return Whether the button was released since the last check.
+   */
+  boolean getButtonReleased(Button button) {
+    return getRawButtonReleased(getButtonIndex(button));
   }
 
-  @Override
-  public void setRumble(RumbleType type, double value) {
-    if (value < 0) {
-      value = 0;
-    } else if (value > 1) {
-      value = 1;
-    }
-    if (type == RumbleType.kLeftRumble) {
-      m_leftRumble = (short) (value * 65535);
-    } else {
-      m_rightRumble = (short) (value * 65535);
+  /**
+   * Returns button index corresponding to a button type.
+   */
+  int getButtonIndex(Button button) {
+    switch (button) {
+      case kBumperLeft:
+        return 5;
+      case kBumperRight:
+        return 6;
+      case kStickLeft:
+        return 9;
+      case kStickRight:
+        return 10;
+      case kA:
+        return 1;
+      case kB:
+        return 2;
+      case kX:
+        return 3;
+      case kY:
+        return 4;
+      case kBack:
+        return 7;
+      case kStart:
+        return 8;
+      default:
+        return 0;
```

Comment:
Instead of making this a switch statement, can we store the value as a field of the enum?